test/e2e/testsuites/specs.go

```diff
@@ -381,10 +381,10 @@ func VerifySuccessfulReplicaAzVolumeAttachments(pod PodDetails, azDiskClient *az
 
 			for _, replica := range replicaAttachments.Items {
 				if replica.Status.State != "Attached" {
```

Comment:
This test fails occasionally because on attachment failure we recreate the azvolumeattachmnet to retry the operation and  as a result we can have more attachments than expected. Instead of getting all replicas for the disk, we should add a FieldSelector and only get the replicas in Attached state to reduce the amount of processing we do here.

Reply:
I don't think this is a big issue. First, failed replica attachment should get deleted in a short amount of time as it involves no cloud operation. That said, it would most likely get deleted before its replacement replica attachment get attached. So, given the 10 minutes timeout, we should see the number of replicas converge and I think leaving it as is would also allow us to test whether it converges or not.

Reply:
Previously, we were seeing the tests fail despite the replicas number converging because the variable `allReplicasAttached` was not being reset after a failed poll. So with the change, we should see success once the replica numbers converge.